tests/zfs-tests/tests/functional/cache/cache_010_pos.ksh

```diff
@@ -56,29 +56,29 @@ function cleanup_testenv
 	fi
 }
 
-log_assert "Cache device can only be block devices."
+log_assert "Cache device can be files, character devices or block devices."
 log_onexit cleanup_testenv
 
 TESTVOL=testvol1$$
 dsk1=${DISKS%% *}
 log_must zpool create $TESTPOOL ${DISKS#$dsk1}
-
 # Add nomal ${DEV_RDSKDIR} device
 log_must zpool add $TESTPOOL cache \
     ${DEV_RDSKDIR}/${dsk1}${SLICE_PREFIX}${SLICE0}
 log_must verify_cache_device $TESTPOOL $dsk1 'ONLINE'
 
 # Add normal file
-log_mustnot zpool add $TESTPOOL cache $VDEV2
+log_must zpool add $TESTPOOL cache $VDEV2
 
 # Add /dev/rlofi device (allowed under Linux)
 if is_linux; then
 	lofidev=$(losetup -f)
 	lofidev=${lofidev##*/}
+	log_must zpool remove $TESTPOOL ${VDEV2%% *}
 	log_must losetup $lofidev ${VDEV2%% *}
 	log_must zpool add $TESTPOOL cache $lofidev
 	log_must zpool remove $TESTPOOL $lofidev
-	log_must losetup -d $lofidev
+	log_must losetup -d "/dev/$lofidev"
```

Comment:
Also differs from the original diff - potentially caused by a change in util-linux versions, however, this should be backwards compatible.